<launch>
  <!-- 카메라 기반 YOLO 감지 (C++) -->
  <node pkg="carvision"
        type="obstacle_detector_car"
        name="obstacle_detector_car"
        output="screen">

    <param name="camera_topic" value="/image_jpeg/compressed"/>
    <param name="model_path"   value="$(find carvision)/scripts/perception/yolov8n.onnx"/>
    <rosparam param="target_class_ids">[2]</rosparam>
    <param name="avoid_height_thresh" value="30.0"/>
    <param name="avoid_x_thresh"      value="350.0"/>
    <param name="confidence_thresh"   value="0.3"/>
    <param name="nms_threshold"       value="0.45"/>
  </node>

  <!-- Pure Pursuit + 회피 노드 (C++) -->
  <node pkg="carvision"
        type="avoid_camera_car"
        name="avoid_camera_car"
        output="screen">

    <param name="ref_file"   value="/root/ws/src/carvision/map/ref.txt"/>
    <param name="path_file"  value="/root/ws/src/carvision/map/Path.txt"/>
    <param name="wheelbase"  value="3.0"/>
    <param name="lookahead"  value="4.5"/>
    <param name="target_kmh" value="15.0"/>
    <param name="avoid_kmh"  value="8.0"/>
    <param name="bbox_height_thresh" value="30.0"/>
    <param name="bbox_x_thresh"      value="350.0"/>
    <param name="bbox_left_thresh"   value="290.0"/>
  </node>
</launch>
